reader/views.py

```diff
@@ -403,15 +410,29 @@ def make_panel_dict(oref, versionEn, versionHe, filter, versionFilter, mode, **k
         if settings_override:
             panel["settings"] = settings_override
         if mode != "Connections" and oref != None:
-            try:
-                text_family = TextFamily(oref, version=panel["currVersions"]["en"], lang="en", version2=panel["currVersions"]["he"], lang2="he", commentary=False,
-                                  context=True, pad=True, alts=True, wrapLinks=False, translationLanguagePreference=kwargs.get("translationLanguagePreference", None)).contents()
-            except NoVersionFoundError:
-                text_family = {}
-            text_family["updateFromAPI"] = True
-            text_family["next"] = oref.next_section_ref().normal() if oref.next_section_ref() else None
-            text_family["prev"] = oref.prev_section_ref().normal() if oref.prev_section_ref() else None
-            panel["text"] = text_family
+            primary_params = [primaryVersion['languageFamilyName'], primaryVersion['versionTitle']]
+            primary_params[0] = primary_params[0] or 'primary'
+            translation_params = [translationVersion['languageFamilyName'], translationVersion['versionTitle']]
+            translation_params[0] = translation_params[0] or 'translation'
+            text_adapter = TextRequestAdapter(oref.section_ref(), [primary_params, translation_params], return_format='wrap_all_entities')
+            text = text_adapter.get_versions_for_query()
+            # text['ref'] = oref.normal()
+            #now we we should add the he and text attributes
```

Comment:
this should be its own function so that the logic is scopes and purpose nicely articulated in the function name